lib/utils/cli.go

```diff
@@ -374,19 +374,45 @@ func EscapeControl(s string) string {
 	return s
 }
 
-// AllowNewlines escapes all ANSI escape sequences except newlines from string and returns a
-// string that is safe to print on the CLI. This is to ensure that malicious
-// servers can not hide output. For more details, see:
-//   - https://sintonen.fi/advisories/scp-client-multiple-vulnerabilities.txt
-func AllowNewlines(s string) string {
-	if !strings.Contains(s, "\n") {
-		return EscapeControl(s)
+// isAllowedWhitespace is a helper function for cli output escaping that returns
+// true if a given rune is a whitespace character and allowed to be unescaped.
+func isAllowedWhitespace(r rune) bool {
+	switch r {
+	case '\n', '\t', '\v':
+		// newlines, tabs, vertical tabs are allowed whitespace.
+		return true
 	}
-	parts := strings.Split(s, "\n")
-	for i, part := range parts {
-		parts[i] = EscapeControl(part)
+	return false
+}
+
+// AllowWhitespace escapes all ANSI escape sequences except some whitespace
+// characters (\n \t \v) from string and returns a string that is safe to
+// print on the CLI. This is to ensure that malicious servers can not hide
+// output. For more details, see:
+//   - https://sintonen.fi/advisories/scp-client-multiple-vulnerabilities.txt
+func AllowWhitespace(s string) string {
+	// loop over string searching for part to escape followed by allowed char.
+	// example: `\tabc\ndef\t\n`
+	// 1. part: ""    sep: "\t"
+	// 2. part: "abc" sep: "\n"
+	// 3. part: "def" sep: "\t"
+	// 4. part: ""    sep: "\n"
+	var sb strings.Builder
+	for i := 0; i < len(s); i++ {
```

Comment:
```suggestion
        // note that increment also happens at bottom of loop because we can safely jump to place where allowedWhitespace was found.
	for i := 0; i < len(s); i++ {
```